### PR TITLE
SWATCH-1956: UMB Offering ingestion should write metered y/n flag in Offering table

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: [SWATCH-1956](https://issues.redhat.com/browse/SWATCH-1956)
Related to [SWATCH-1823](https://issues.redhat.com/browse/SWATCH-1823)
Related to [SWATCH-1840](https://issues.redhat.com/browse/SWATCH-1840)

## Description
- We then have to store a new column on the Offering table. "Metered" column to differentiate between payg and non-payg.
- The product tag determination can be tested using the  /internal/offerings/{sku}/product_tags endpoint. 
- Remove usage of offering.product name.

## Testing
 1. Add any SKU with product name as "RHEL Server" and could have billing_account_id from AWS ex: MCT3557HR - Red Hat Enterprise Linux, Premium (On-Demand, 1 Virtual Node, Hourly, Billing) with billing account id and billing provider.

(SKU addition can be done via endpoint https://mtls.internal.cloud.stage.redhat.com/app/rhsm-capacity-ingress/internal/api/rhsm-subscriptions/v1/internal/subscriptions?reconcileCapacity=true )

from smqe_tools import internal_api_add_sku

internal_api_add_sku(account_number="11490832", sku="MCT3557HR", org_id=app.user.identity.org_id, start_day="2023-10-01", end_day="2023-10-30", billing_provider="aws", billing_account_id="9458475893")

2. Sync capacity.

3. Verify SKU is displayed under rhel-for-x86-eus-payg

### CODE REVIEW

File: src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java [hunk 1]
```diff
@@ -263,19 +263,14 @@ private boolean ensureOffering(
 
   private void checkForMissingBillingProvider(
       org.candlepin.subscriptions.db.model.Subscription subscription) {
-    if (subscription.getBillingProvider() == null
-        || subscription.getBillingProvider().equals(BillingProvider.EMPTY)) {
+    if ((subscription.getBillingProvider() == null
+            || subscription.getBillingProvider().equals(BillingProvider.EMPTY))
+        && subscription.getOffering().isMetered()) {
       // The offering here is going to be a proxy object created by getReferenceById.  Hibernate
       // should take care of actually performing the select from the database if one is needed.
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java:270]: Deleted

File: src/main/resources/liquibase/202312200751-add-metered-column-for-offering-table.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.20.xsd">
+  <changeSet id="202312200751-01" author="jcarvaja">
+    <comment>Add column for metered column to the offering table</comment>
+    <addColumn tableName="offering">
+      <column name="metered" type="BOOLEAN"/>
+    </addColumn>
+    <rollback>
+      <dropColumn tableName="offering" columnName="metered"/>
+    </rollback>
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply src/main/resources/liquibase/202312200751-add-metered-column-for-offering-table.xml:14]: Good to know, thanks!

**Comment by Sgitario** [comment]
@awood PR updated with your comments. Many thanks!

**Comment by awood** [comment]
Just as a note, I did an offering sync from stage and all ~15,000 products were marked as "false" for the metered value.  I'm not sure if that's something we need to follow-up with them or not.

**Review by awood — APPROVED** [review]

**Review by TrayvonMcKnight — APPROVED** [review]

**Pull request merged**